File: server/schemas/coach_summary.py
from __future__ import annotations

from typing import Literal, List, Optional

from pydantic import BaseModel, Field

from server.schemas.coach_diagnosis import CoachDiagnosis
from server.schemas.player_profile import PlayerModel


class CoachSgCategory(BaseModel):
    name: Literal["tee", "approach", "short", "putt"]
    sg: float


class CoachHoleSg(BaseModel):
    hole: int
    gross_score: int
    sg_total: float
    worst_category: Optional[str] = None


class CoachSequenceSummary(BaseModel):
    max_shoulder_rotation: float
    max_hip_rotation: float
    max_x_factor: float
    sequence_order: List[str] = Field(default_factory=list)
    is_ideal: bool


class CoachCaddieHighlight(BaseModel):
    trusted_club: Optional[str] = None
    trusted_club_trust_score: Optional[float] = None
    ignored_club: Optional[str] = None
    ignored_club_trust_score: Optional[float] = None


class CoachMissionSummary(BaseModel):
    mission_id: Optional[str] = None
    mission_label: Optional[str] = None
    success: Optional[bool] = None


class CoachRoundSummary(BaseModel):
    run_id: str
    member_id: Optional[str] = None
    course_name: Optional[str] = None
    tees: Optional[str] = None
    date: Optional[str] = None
    score: Optional[int] = None

    sg_total: Optional[float] = None
    sg_by_category: List[CoachSgCategory] = Field(default_factory=list)
    sg_per_hole: List[CoachHoleSg] = Field(default_factory=list)

    sequence: Optional[CoachSequenceSummary] = None
    caddie: Optional[CoachCaddieHighlight] = None
    mission: Optional[CoachMissionSummary] = None
    diagnosis: Optional[CoachDiagnosis] = None
    player_model: Optional[PlayerModel] = None

File: web/src/api/coachSummary.ts
import axios from "axios";

import { API, withAuth } from "@/api";

export type CoachCategory = "tee" | "approach" | "short" | "putt" | "sequence" | "strategy";

export type CoachFinding = {
  id: string;
  category: CoachCategory;
  severity: "info" | "warning" | "critical";
  title: string;
  message: string;
  evidence?: Record<string, unknown>;
  suggested_missions?: string[];
  suggested_focus?: string[];
};

export type CoachDiagnosis = {
  run_id: string;
  findings: CoachFinding[];
};

export type CoachSgCategory = {
  name: "tee" | "approach" | "short" | "putt";
  sg: number;
};

export type CoachHoleSg = {
  hole: number;
  gross_score: number;
  sg_total: number;
  worst_category?: string | null;
};

export type CoachSequenceSummary = {
  max_shoulder_rotation: number;
  max_hip_rotation: number;
  max_x_factor: number;
  sequence_order: string[];
  is_ideal: boolean;
};

export type CoachCaddieHighlight = {
  trusted_club?: string | null;
  trusted_club_trust_score?: number | null;
  ignored_club?: string | null;
  ignored_club_trust_score?: number | null;
};

export type CoachMissionSummary = {
  mission_id?: string | null;
  mission_label?: string | null;
  success?: boolean | null;
};

export type CoachRoundSummary = {
  run_id: string;
  member_id?: string | null;
  course_name?: string | null;
  tees?: string | null;
  date?: string | null;
  score?: number | null;
  sg_total?: number | null;
  sg_by_category: CoachSgCategory[];
  sg_per_hole: CoachHoleSg[];
  sequence?: CoachSequenceSummary | null;
  caddie?: CoachCaddieHighlight | null;
  mission?: CoachMissionSummary | null;
  diagnosis?: CoachDiagnosis | null;
};

export async function fetchCoachRoundSummary(
  runId: string,
): Promise<CoachRoundSummary> {
  const response = await axios.get<CoachRoundSummary>(
    `${API}/api/coach/round-summary/${runId}`,
    { headers: withAuth() },
  );
  return response.data;
}

export async function fetchCoachDiagnosis(runId: string): Promise<CoachDiagnosis> {
  const response = await axios.get<CoachDiagnosis>(
    `${API}/api/coach/diagnosis/${runId}`,
    { headers: withAuth() },
  );
  return response.data;
}
